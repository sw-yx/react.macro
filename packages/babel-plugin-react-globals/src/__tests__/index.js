import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '../'

const projectRoot = path.join(__dirname, '../../')

expect.addSnapshotSerializer({
  print(val) {
    return val.split(projectRoot).join('<PROJECT_ROOT>/')
  },
  test(val) {
    return typeof val === 'string'
  },
})
const fixture = filename => ({
  fixture: require.resolve(`./fixtures/${filename}`),
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {filename: __filename},
  tests: {
    'basic test of functionality': fixture('basic'),
  },
})
