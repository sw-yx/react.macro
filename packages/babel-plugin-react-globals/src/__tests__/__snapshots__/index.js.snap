// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-globals basic test of functionality: basic test of functionality 1`] = `

import React from 'react.macro'

const CTX = createContext(4)
const Hi = () => {
  const state = useState(4)
  return <div> hi </div>
}

class App extends Component {
  render() {}
}

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react.macro';

const CTX = React.createContext(4);
const Hi = () => {
  const state = React.useState(4);
  return <div> hi </div>;
};

class App extends React.Component {
  render() {}
}

`;
